core/shape-parts.cpp

```diff
@@ -39,6 +39,32 @@ const Path& BasicBezier::generate(Point endPoint, Point transitionPoint) {
   return m_path;
 }
 
+const Path& FlatBezier::generate(Point endPoint, Point transitionPoint) {
+  MCFixed flatEndY = m_flatWidth / -2; // WHY DO I HAVE TO MAKE THIS NEGATIVE TO GET IT TO WORK???
```

Comment:
I haven't documented how these parts work clearly. I'll work on updating the the wiki and add a link here when I'm done. In a nutshell, it works like this: the nose is at 0,0 on the left, and the tail is at length,0 on the right. These parts build the upper half of the shape only; we mirror that to get the lower half. Thus, all Y coordinates should be positive. For the nose portion (`endPoint.X < transitionPoint.X`), imagine drawing the shape starting at the center of the nose and proceeding clockwise from 9pm to midnight. On the tail, it goes from midnight to 3am. Here you should have `flatEndY = m_flatWidth / 2` for sure. I'll make notes for you below on how to adjust the rest.